### PR TITLE
Release 2.0.0 to support vending multiple tokens and dashboard

### PR DESCRIPTION
**Pull request by avipinku**
**Description of changes:**

* Create and vend multiple tokens instead of just RW (RO, RW, admin)
* Various cleanup and bug fixes
* Increased testing coverage, from 63 -> 76
* This is NOT backwards compatible (new version 1.1.0)

**Why is this change necessary:**
* To support a one-click dashboard
**How was this change tested:**
* End-to-end deployments with Grafana in various configurations (different ports, http/https, etc.)
**Any additional information or context required to review the change:**

**Checklist:**
- [x] Updated the README if applicable
- [x] Updated or added new unit tests
- [ ] Updated or added new integration tests
- [ ] Updated or added new end-to-end tests
- [ ] If your code makes a remote network call, it was tested with a proxy
- [ ] You confirm that the change is backwards compatible

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/influxDBTokenPublisher.py [hunk 1]
```diff
@@ -19,6 +19,9 @@
 
 logging.basicConfig(level=logging.INFO)
 TIMEOUT = 10
+# Influx commands need to be given the port of InfluxDB inside the container, which is always 8086 unless
+# overridden inside the InfluxDB config
```
[the remark this answers is not in the document]
Comment (avipinku) [reply src/influxDBTokenPublisher.py:23]: So it does not need to be overridden - customers should not need to change that port at all, it is just a virtual port mapped inside the container to the port in the config value. We just need to map it to the requested port on the host.

File: src/influxDBTokenStreamHandler.py [hunk 2]
```diff
@@ -40,11 +41,11 @@ def handle_stream_event(self, event: SubscriptionResponseMessage) -> None:
         """
         try:
             message = str(event.binary_message.message, "utf-8")
```
[the remark this answers is not in the document]
Comment (avipinku) [reply src/influxDBTokenStreamHandler.py]: fixed

File: recipe.yaml [hunk 3]
```diff
@@ -15,7 +15,7 @@ ComponentConfiguration:
   DefaultConfiguration:
     AutoProvision: 'true'
     InfluxDBMountPath: '/home/ggc_user/dashboard'
-    SecretArn: 'arn:aws:secretsmanager:<region>:<account>:secret:<name>'
+    SecretArn: 'arn:aws:secretsmanager:region:account:secret:name'
```
Comment (pallavi03) [inline recipe.yaml:18]: why is this changed?

Comment (avipinku) [reply recipe.yaml:18]: To avoid potential parsing errors with the "<" ">" characters

**Review by pallavi03 — DISMISSED** [review]

File: src/influxdb_utils.sh [hunk 4]
```diff
@@ -6,24 +6,31 @@
...
   RETRIES=0
   until [ "$CONTAINER_SETUP_STATUS" == "OK" ] || [ "$RETRIES" -eq 4 ]; do
     sleep 10
     echo "Attempt $RETRIES: Waiting until InfluxDB reports a status of OK..."
-    if [ $SERVER_PROTOCOL == "http" ]; then
-      CONTAINER_SETUP_STATUS=$(docker exec $CONTAINER_NAME influx ping) || true
-    elif [ $SERVER_PROTOCOL == "https" ]; then
-      CONTAINER_SETUP_STATUS=$(docker exec $CONTAINER_NAME influx ping --host "https://$CONTAINER_NAME:$INFLUXDB_PORT" --skip-verify) || true
+    if [ "$SERVER_PROTOCOL" == "http" ]; then
+      CONTAINER_SETUP_STATUS=$(docker exec "$CONTAINER_NAME" influx ping --host "http://$CONTAINER_NAME:8086") || true
```
Comment (saranyailla) [inline src/influxdb_utils.sh:31]: Why are we hard-coding 8086? Is it fixed?

File: recipe.yaml (hunk 3 above)
Comment (saranyailla) [reply recipe.yaml:18]: When is this a problem?

File: src/influxdb_utils.sh (hunk 4 above)
Comment (avipinku) [reply src/influxdb_utils.sh:31]: Yes - 8086 here is the port inside the container, which we map to a configurable port

File: recipe.yaml (hunk 3 above)
Comment (avipinku) [reply recipe.yaml:18]: I've seen it happen during the secret retrieval process- but this is just a placeholder anyway, and is expected to be replaced

File: src/influxDBTokenStreamHandler.py [hunk 5]
```diff
@@ -80,13 +81,51 @@ def on_stream_closed(self) -> None:
...
+        Parse the correct token based on the IPC message received, and construct the final JSON to publish.
+
+        :param message: the received IPC messsage
+        :return: the complete JSON, including token, to publish
+        """
+
+        loaded_token_json = json.loads(self.influxDB_token_json)
+        publish_json = json.loads(self.influxDB_metadata_json)
+
+        if message['action'] == 'RetrieveToken':
```
[the remark this answers is not in the document]
Comment (avipinku) [reply src/influxDBTokenStreamHandler.py:95]: Re: our discussion offline:

it should always exist, since it’s constructed and sent by the influxdbpublisher. If it doesn’t exist, `getpublishjson()` will throw an exception that is caught by the caller, `handle_stream_event()`

**Review by pallavi03 — DISMISSED** [review]

File: recipe.yaml [hunk 6]
```diff
@@ -1,7 +1,7 @@
 ---
 RecipeFormatVersion: '2020-01-25'
 ComponentName: aws.greengrass.labs.database.InfluxDB
-ComponentVersion: '1.0.0'
+ComponentVersion: '1.1.0'
```
Comment (saranyailla) [inline recipe.yaml:4]: Because this is backwards incompatible, should we call it 2.0.0 instead of just minor version bump.

**Review by saranyailla — APPROVED** [review]

File: src/influxDBTokenStreamHandler.py [hunk 7]
```diff
@@ -80,13 +81,51 @@ def on_stream_closed(self) -> None:
...
+            token = ''
+            if message['accessLevel'] == 'RW':
+                token = next(d for d in loaded_token_json if d['description'] == 'greengrass_readwrite')['token']
+            elif message['accessLevel'] == 'RO':
+                token = next(d for d in loaded_token_json if d['description'] == 'greengrass_read')['token']
+            elif message['accessLevel'] == 'Admin':
+                # Admin token description is in the format "USERNAME's Token"
+                if "'s Token" in loaded_token_json[0]['description']:
```
Comment (saranyailla) [inline src/influxDBTokenStreamHandler.py:103]: Nit: Extract the string as `admin_token_identifier= "'s Token"`

File: src/influxDBTokenStreamHandler.py (hunk 5 above)
Comment (saranyailla) [inline src/influxDBTokenStreamHandler.py:95]: Nit: Could reduce the  complexity by using `not` here  
```
if not message['action'] == 'RetrieveToken':
   return None
```

and 
```
if not "'s Token" in loaded_token_json[0]['description']:
   return None
```

**Review by saranyailla — DISMISSED** [review]

**Review by pallavi03 — DISMISSED** [review]

File: recipe.yaml (hunk 6 above)
Comment (avipinku) [reply recipe.yaml]: Fixed

**Review by saranyailla — APPROVED** [review]

**Review by pallavi03 — APPROVED** [review]
